apiserver/facades/client/bundle/mock_test.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package bundle_test
+
+import (
+	"github.com/juju/juju/apiserver/facades/client/bundle"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/testing"
+)
+
+type mockState struct {
+	testing.Stub
+	bundle.Backend
+	str params.StringResult
+}
+
+func (m *mockState) ExportBundle() (params.StringResult, error) {
```

Comment:
The method we need here is the state method Export() - that is what we are mocking.
The idea is that the facade calls a state object to do stuff. In production it is the real state. In tests, it is a mock which must have the same interface as the real state. So mockState needs to implement migration.StateExporter. 
Also, don't embed Backend interface in mockState as that just masks the fact that it hasn't been implemented properly.
You just want
type mockState struct {
testing.Stub
}